packages/notifier/src/asyncIteratableKit.js

```diff
@@ -0,0 +1,84 @@
+// @ts-check
+// eslint-disable-next-line spaced-comment
+/// <reference types="ses"/>
+
+import { HandledPromise, E } from '@agoric/eventual-send';
+import { makePromiseKit } from '@agoric/promise-kit';
+
+import './types';
+
+/**
+ * This makes a pair of an `updater` and an initial `asyncIteratable`. The
+ * `updater` API is the same as the Notifier's `updater`. In both cases,  calls
+ * to the `updater` produce a stream of non-final values terminated with
+ * either a final success value or a final failure reason. The purpose of the
+ * notifier is to be lossy over the non-final values. The purpose of the
+ * asyncIterableKit is to consume such streams losslessly.
```

Comment:
Order of last two sentences confused me about the purpose of this. Let’s separate the definition from the relationship to other tools more cleanly.

Some ideas:

> This makes an entangled lossless async updater and iterator pair. The updater may over time produce any number of non-final values, terminated with either a final successful completion value or failure reason.
>
> The async updater has the same interface as the related notifier. However, this function ensures that all values will transit in order without dropping any intermediate values.

This differs from my sketch of `pipe` in that the methods of the updater do not return promises to optionally synchronize the producer and consumer pace. https://gist.github.com/kriskowal/abcba770b5dba526a1af801d692fd94e

Reply:
Documentation significantly changed. PTAL

Reply:
Enough docs that I'm resolving this. But please PTAL anyway.